### PR TITLE
refact(rpc): merge rpc module into commons

### PR DESCRIPTION
**Pull request by imbajin**
for apache [need](https://github.com/hugegraph/hugegraph/issues/1635), after this pr, we will:
- [x] rename hugegraph-common --> hugegraph-commons
- [x] merge the hugegraph-rpc module into hugegraph-commons, and keep it independent (include commit log)
- [x] merge License, checkstyle, ci, etc..
- [x] add a new `pom.xml` & `README` in root dir 
- [ ] unify rpc & common version later (better to keep same)

### CODE REVIEW

File: .github/workflows/stale.yml [hunk 1]
```diff
@@ -20,17 +20,19 @@ jobs:
...
         operations-per-run: 10
-        start-date: '2018-12-01T00:00:00Z'
+        start-date: '2018-10-01T00:00:00Z'
 
         exempt-all-assignees: true
         remove-stale-when-updated: true
         exempt-all-pr-milestones: true
         delete-branch: false
-        enable-statistics: true
+        enable-statistics: true
```
Comment (javeme) [inline .github/workflows/stale.yml:38]: expect blank line

**Comment by imbajin** [comment]
after review & CI pass, separate this pr into 2

**Review by javeme — DISMISSED** [review]
LGME

File: hugegraph-rpc/pom.xml [hunk 2]
```diff
@@ -3,49 +3,30 @@
...
-            <distribution>repo</distribution>
-        </license>
-    </licenses>
-
-    <scm>
-        <url>https://github.com/hugegraph/hugegraph-rpc</url>
-        <connection>https://github.com/hugegraph/hugegraph-rpc</connection>
-        <developerConnection>https://github.com/hugegraph/hugegraph-rpc</developerConnection>
-    </scm>
+    <description>HugeGraph Database RPC component</description>
```
Comment (javeme) [inline hugegraph-rpc/pom.xml:18]: need to update https://github.com/hugegraph/hugegraph-rpc link?

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+            <email>javaloveme@gmail.com</email>
+        </developer>
+        <developer>
+            <name>zhoney</name>
+            <email>zhangyi89817@126.com</email>
+        </developer>
+        <developer>
+            <name>liningrui</name>
+            <email>liningrui@vip.qq.com</email>
+        </developer>
```
Comment (javeme) [inline pom.xml]: can add more developers

Comment (imbajin) [reply pom.xml]: it's optional & maybe do it in a separate pr (if need)

File: hugegraph-rpc/pom.xml (hunk 2 above)
Comment (imbajin) [reply hugegraph-rpc/pom.xml:18]: like change to `https://github.com/hugegraph/hugegraph-commons/hugegraph-rpc` ?

Comment (javeme) [reply hugegraph-rpc/pom.xml:18]: it's ok to just delete the url option

**Review by javeme — APPROVED** [review]

**Review by zhoney — APPROVED** [review]

**Review by javeme — APPROVED** [review]

**Pull request merged**